src/reporters/udp_sender.js

```diff
@@ -120,23 +127,24 @@ export default class UDPSender {
             return {err: false, numSpans: 0}
         }
 
-        let bufferLen = this._byteBufferSize + this._emitSpanBatchOverhead;
+        let bufferLen = this._totalSpanBytes + this._emitSpanBatchOverhead;
         let thriftBuffer = new Buffer(bufferLen);
-        let bufferResult = this._agentThrift.Agent.emitBatch.argumentsMessageRW.writeInto(
+        let writeResult = this._agentThrift.Agent.emitBatch.argumentsMessageRW.writeInto(
             this._convertBatchToThriftMessage(this._batch), thriftBuffer, 0
         );
 
-        if (bufferResult.err) {
-            console.log('err', bufferResult.err);
+        if (writeResult.err) {
+            this._logger.error(`error writing Thrift object: ${writeResult.err}`);
```

Comment:
Do we want to emit a metric here?

Reply:
the sender is not integrated with metrics currently. Also the two new errors are a sign of misconfiguration, not something that can keep happening repeatedly, thus I think the logs are a lot more useful.